### DISCUSSION TITLE
Adding a "correct response section in the data for button responses

### DISCUSSION THREAD

**Discussion by rbruno1** [jspsych/jsPsych#1941] [Q&A] [answered]
here is the code for one of my trials
'''const Buck_trial_1 = {
  type: 'audio-button-response',
  stimulus: 'BucketThere2_POD_AO_Frames2.wav',
  choices: ['Bucket', 'Buckle'],
  post_trial_gap: 1000,
  data: {condition: 'Bucket', correct_response: '0'},
  on_finish: function(data) {
    data.correct = jsPsych.pluginAPI.compareKeys(data.response, data.correct_response);
  }
};
''"
I am trying to get it to show the "correct response" in the display data section, but the responses of the participants are recorded as 0 or 1, while the correct response are defined as '0' or '1'. how do i get the correct_response function to recognize the responses of the participant?

**Comment by rbruno1** [comment, marked as answer]
correct_response: '0' --> correct_response: 0

**Comment by HannaKdeSouza** [comment]
I had the same problem. Didn't know how to get the data to show the correct response as it showed everything as 'false'. It was solved by removing the ' ' from the correct responses. Thanks for the help!:

var timeline_variables = [
    {sound: "bead_female1_man.wav", 0: ['bead'], 1:['bid'], correct_response: 0},
    {sound: "seed_male1_man.wav", 0: ['seed'], 1:['Sid'], correct_response: 0},

var response_trial = {
type: 'html-button-response',
stimulus: '',
choices: [jsPsych.timelineVariable ('0'), jsPsych.timelineVariable('1')],
prompt: "<p>Click on the word you heard</p>",
trial_duration: 6000,
margin_vertical: '40px',
margin_horizontal: '40px',
response_ends_trial: true,
post_trial_gap: 500,
data: {
    phase: 'response',
    sound: jsPsych.timelineVariable('sound'), 
    correct_response: jsPsych.timelineVariable('correct_response')
},
on_finish: function(data){
    data.correct=jsPsych.pluginAPI.compareKeys(data.response, data.correct_response)
    }

**Comment by becky-gilbert** [comment]
Just to clarify this for anyone else who has the same problem: as of jsPsych v6.3.0, the button-response plugins now store the button that was pressed in a data property called `response`, which is a number (not a string). So when you create the correct response value for each trial, that value should be a number as well. 

Also, note that there's no need to use the `compareKeys` function for comparing two button responses. Instead you can just do `data.correct = data.response === data.correct_response`. See my comment about that [here](https://github.com/jspsych/jsPsych/discussions/2114#discussioncomment-1273767).

**Comment by HannaKdeSouza** [comment]
Thank you! To be honest, I had no idea how to get the correct response for a button-response trial as all the documents I found were for html-keyboard-response trials. So great that this was also clarified!
